### PR TITLE
Add sqrt_stretch and custom mask to align steps

### PR DESCRIPTION
**Pull request by wbalmer**
See discussion in #114 . For data with more complex speckle patterns (e.g. NIRCam bar NARROW coronagraphy), it can be necessary to emphasize the speckle pattern itself rather than the central leakage term. I've found a combination of two strategies reproduces the commanded shifts from the SGD: masking the central leakage using a custom mask during the `align_frames` step, and taking the square root of the images before cross correlation. This PR introduces non-intrusive toggles to do both of these things.

### CODE REVIEW

**Comment by wbalmer** [comment]
Need to update tests with this

**Comment by wbalmer** [comment]
Hi Aarynn. I see your point about generalizing/simplifying the statement. I suppose the problem is that np.power, the function I'd use to generalize this, returns nans when negative values (which there will be in median subtracted data) are raised to non integer exponents (like 1/2). np.sqrt and np.cbrt provides an interface for returning the principle root. So to implement something like you describe, I feel like it would end up being about as many if/else statements as there are currently (a case for exp=1/2 or 1/3 and then a generalized case for exp>=1). I can do this, if you'd still like to see the generalization

**Comment by AarynnCarter** [comment]
@wbalmer Okay gotcha, in that case maybe we can strike a middle ground. For the nominal case (exp=1), we could use your current else statement

```
else:
    img1 = datasub* masksub
    img2 = model_psf* masksub
```

But for anything else, we can adjust to the following:

```
if take_sqrt:
    img1 = np.sqrt(np.abs(datasub))* masksub
    img2 = np.sqrt(np.abs(model_psf))* masksub
```

to 

```
if scale_exponent != 1:  #Or some other sensible name
    img1 = np.power(np.abs(datasub), scale_exponent)* masksub
    img2 = np.power(np.abs(model_psf), scale_exponent)* masksub
```

I feel like this keeps us to the current number of if statements, but more general, and avoids using the absolute function for the nominal case. For exponents >1, it won't matter that we've taken the absolute value as the result will be the same. Do you think that would make sense?

**Comment by wbalmer** [comment]
Sounds good! I'll test this now.

**Comment by wbalmer** [comment]
Okay, tested in both the shft_exp = 1 and shft_exp = 0.5 cases and expected behavior was reproduced. Should be good to go

File: spaceKLIP/imagetools.py [hunk 1]
```diff
@@ -1919,9 +1926,16 @@ def find_nircam_centers(self,
                                     xycen=(xc, yc),
                                     npix=fov_pix)
             
+            if shft_exp != 1:
+                img1 = datasub* masksub
+                img2 = model_psf* masksub
+            else:
+                img1 = np.power(datasub, shft_exp)* masksub
```
Comment (AarynnCarter) [inline spaceKLIP/imagetools.py:1933]: I think the abs() functions are missing here?

**Review by AarynnCarter — CHANGES_REQUESTED** [review]
Just one tweak, but otherwise looks good!

**Comment by wbalmer** [comment]
Should be fixed!

**Review by AarynnCarter — APPROVED** [review]

**Pull request merged**